File: masteringNodejs/speed-loop.py
# Python looped 1000000000 times in 34.590 seconds on osx
import time

def main() :

  cycles = 1000000000
  start = time.perf_counter()

  for i in range (0, cycles) :
    pass # Empty Loop 

  end = time.perf_counter()
  duration = end -start
 
  print("Python looped %d times in %.3f seconds" % (cycles, duration))

main()

File: masteringNodejs/speed-loop.c
/* 
 * C looped 1000000000 times in 2.576541 second on osx 
 * C looped 1000000000 times in 1.660506 seconds on Windows WSL
*/
#include <stdio.h>
#include <time.h>
int main () {
  int cycles = 1000000000;
  clock_t start,end;
  double duration;
  start = clock();
  for (int i = 0 ; i < cycles ; i++) {
	/* Empty Loop */ 
        }
  end = clock();
  duration = ((double)(end -start)) / CLOCKS_PER_SEC;
  printf("C looped %d times in %lf seconds\n", cycles, duration);
 return 0;
}   
